source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,100 @@
+const tippy = require('tippy.js').default
+const { ipcRenderer } = require('electron')
+
+/**
+ * A hook for displaying link tooltips which can be used to visually
+ * enable users to click a link (without having to press down Ctrl/Cmd)
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+module.exports = (cm) => {
+  cm.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+
+    // Initialise displayed attributes to 'loading'
+    let title = 'loading'
+    let content = 'loading'
+    let wordCount = 'loading'
+    let time = 'loading'
+    // Create a tippy. This will display the loading values
+    let tooltip = tippy(a, {
+      content: 'Searching For File...',
+      allowHTML: true, // Obviously
+      interactive: true,
+      placement: 'top-start', // Display at the beginning of the anchor
+      appendTo: document.body, // anchor
+      showOnCreate: true, // Immediately show the tooltip
+      arrow: false, // No arrow for these tooltips
+      onHidden (instance) {
+        instance.destroy() // Destroy the tippy instance.
+      }
+    })
+
+    // Find the file's absolute path
+    ipcRenderer.invoke('application', { command: 'file-path-find', payload: a.innerText })
+      .then((filepath) => {
```

Comment:
I would refactor so that the callback just calls a different function with the loaded file contents. Plus I just saw you're invoking two asynchronous calls in succession, and that's not only callback hell but a hell to maintain. I don't have an ad-hoc solution, but the good solution will definitely use functions and a better state management. Mh … we have to get to this.